### PR TITLE
fix ams_xfr.rb - NoMethodError undefined method 'empty?' for nil:NilC…

### PR DESCRIPTION
**Pull request by nixawk**
@digininja fix issue https://github.com/rapid7/metasploit-framework/issues/6888
## Verification

List the steps needed to make sure this thing works
- [x] Start `msfconsole`
- [x] `use exploit/windows/antivirus/ams_xfr`
- [x] `set RHOST 192.168.24.1`
- [x] `unset CMD` 
- [x] `run`

```
msf exploit(ams_xfr) > unset CMD
Unsetting CMD...
msf exploit(ams_xfr) > run

[*] Started reverse TCP handler on 192.168.24.139:4444
[*] 192.168.24.1:12174 - Sending request to 192.168.24.1:12174
[-] 192.168.24.1:12174 - Exploit failed: Errno::EACCES Permission denied - bind(2) for [::]:69
[*] Exploit completed, but no session was created.
```

### CODE REVIEW

File: modules/exploits/windows/antivirus/ams_xfr.rb [hunk 1]
```diff
@@ -93,7 +93,7 @@ def execute_command(cmd, opts = {})
 
   def exploit
 
-    if not datastore['CMD'].empty?
+    if not (datastore['CMD'].nil? || datastore['CMD'].empty?)
```
Comment (void-in) [inline modules/exploits/windows/antivirus/ams_xfr.rb]: Or we can use ```.blank?``` which covers both ```nil?``` and ```empty?```
```ruby
if ! datastore['CMD'].blank?
```

Comment (wchen-r7) [inline modules/exploits/windows/antivirus/ams_xfr.rb]: ```unless datastore['CMD'].blank?```

Comment (join-us) [inline modules/exploits/windows/antivirus/ams_xfr.rb]: Thanks @void-in .

**Comment by join-us** [comment]
```
metasploit-framework [fix_issue_6888] grep -R "datastore\['CMD'\].empty?" ./
.//modules/exploits/multi/http/struts_code_exec.rb:    if not datastore['CMD'].empty?
.//modules/exploits/multi/http/struts_code_exec_exception_delegator.rb:    if not datastore['CMD'].empty?
.//modules/exploits/windows/antivirus/ams_hndlrsvc.rb:    if not datastore['CMD'].empty?
.//modules/exploits/windows/http/ca_totaldefense_regeneratereports.rb:    if not datastore['CMD'].empty?
.//modules/exploits/windows/http/osb_uname_jlist.rb:    if not datastore['CMD'].empty?
```

**Comment by join-us** [comment]
@wchen-r7 @void-in I will fix all here.

**exploit/multi/http/struts_code_exec**

```
msf exploit(struts_code_exec) > set URI /index
URI => /index
msf exploit(struts_code_exec) > unset CMD
Unsetting CMD...
msf exploit(struts_code_exec) > run

[*] Started reverse TCP handler on 10.0.250.51:4444
[-] Exploit failed: NoMethodError undefined method `empty?' for nil:NilClass
[*] Exploit completed, but no session was created.
```

**exploit/multi/http/struts_code_exec_exception_delegator**

```
msf exploit(struts_code_exec_exception_delegator) > unset CMD
Unsetting CMD...
msf exploit(struts_code_exec_exception_delegator) > run

[*] Started reverse TCP handler on 10.0.250.51:4444
[-] Exploit failed: NoMethodError undefined method `empty?' for nil:NilClass
[*] Exploit completed, but no session was created.
```

**Comment by wchen-r7** [comment]
ok sure, I'll check all of them. Thanks!

**Comment by sempervictus** [comment]
More blowback from actually fixing the way datastore options are handled (vs the old mistake of "stringification" for all values, nil/empty/numeric/etc). Thanks for cleaning this up.

**Comment by busterb** [comment]
Thanks @join-us & @wchen-r7. Indeed @sempervictus , lots of landmines to defuse!

**Pull request merged**